### ISSUE TITLE
[Infrastructure] Make sure the SDK tests run against an up to date version of the ASP.NET Core runtime

### ISSUE THREAD

**Issue by javiercn** [dotnet/aspnetcore#32279]
https://github.com/dotnet/sdk/blob/a30e465a2e2ea4e2550f319a2dc088daaafe5649/Directory.Build.targets#L42-L46

The tests in the SDK will use the correct runtime version, however they won't do the same for ASP.NET Core. This causes issues where things that break the SDK go unnoticed until ASP.NET Core needs to consume a new SDK version, at which point is blocked from doing so because of the incompatibility that wasn't caught.

**Comment by mkArtakMSFT** [comment]
@pranavkm, you've looked at the short term solution for this and also I believe you had some ideas about next steps here. What were these? Can you please share here? Thanks.

**Comment by pranavkm** [comment]
The plan was to stop running tests against 6.0 / current version of the runtime and instead run it against 3.1.

**Comment by dougbu** [comment]
How does that help test the latest SDK with our latest shared Fx contributions @pranavkm❔ Fix to duplicate https://github.com/dotnet/sdk/blob/a30e465a2e2ea4e2550f319a2dc088daaafe5649/Directory.Build.targets#L42-L46 for Microsoft.AspNetCore.App would be a significant improvement rather than a step back.

**Comment by dougbu** [comment]
@mkArtakMSFT if your team is unable to address this issue, I suggest moving it to the dotnet/sdk repo and requesting help from @marcpopMSFT's team.

**Comment by dougbu** [comment]
@javiercn may I ask why this issue is in dotnet/aspnetcore and not dotnet/sdk❔

**Comment by dougbu** [comment]
@wtgodbe this may be something you could look at while on sdk / installer blops duty. It doesn't have to be done before you leave the rotation of course. Would at least be interesting to see if extending https://github.com/dotnet/sdk/blob/48be854400cb4c05698e7a686e08fdef7cfe20ac/Directory.Build.targets#L46-L54 is enough. May need to change some test infra code as well.

**Comment by wtgodbe** [comment]
@marcpopMSFT is this something the SDK team would consider doing?

**Comment by marcpopMSFT** [comment]
Not sure the targets file will do what you want as that'll be used for building the SDK bits, not for running the tests. What I think you want is an update to this task: https://github.com/dotnet/sdk/blob/main/src/Layout/toolset-tasks/OverrideAndCreateBundledNETCoreAppPackageVersion.cs

You would need to add a BundledNETCoreAppPackageVersion in installer first. Then you'd have to modify code further down to find the aspnet items rather than using First() everywhere. Not sure the SDK team can pick this up but we can certainly help provide guidance for someone who wants to do this.

**Comment by marcpopMSFT** [comment]
And in theory we'd want this updating all the WD ones as well...uggh. CC @dsplaisted
